yb-voyager/cmd/analyzeSchema.go

```diff
@@ -584,32 +575,26 @@ func createSqlStrInfoArray(path string, objType string) []sqlInfo {
 		singleLine += curr + " "
 		singleString += curr + "\n"
 
-		if codeBlock {
-			// Assuming that both the dollar quote strings will not be in same line
-			if dollarQuoteFlag == 0 {
-				if idx := dollarQuoteRegex.FindStringSubmatch(curr); idx != nil {
-					dollarQuoteFlag = 1 //denotes start of the code/body part
-					outerDollarQuotedStr = idx[0]
-				} else if strings.Contains(curr, ";") { // in case, there is no body part
-					//one liner sql string created, now will check for obj count and report cases
-					processCollectedSql(path, &singleLine, &singleString, objType, &sqlInfoArr, &reportNextSql)
-				}
-			} else if dollarQuoteFlag == 1 {
-				if strings.Contains(curr, outerDollarQuotedStr) {
-					dollarQuoteFlag = 2 //denotes end of code/body part
-				}
+		// Assuming that both the dollar quote strings will not be in same line
+		if dollarQuoteFlag == 0 {
+			if idx := dollarQuoteRegex.FindStringSubmatch(curr); idx != nil {
+				dollarQuoteFlag = 1 //denotes start of the code/body part
+				outerDollarQuotedStr = idx[0]
+			} else if strings.Contains(curr, ";") { // in case, there is no body part
+				//one liner sql string created, now will check for obj count and report cases
+				processCollectedSql(path, &singleLine, &singleString, objType, &sqlInfoArr, &reportNextSql)
 			}
-			if dollarQuoteFlag == 2 {
-				if strings.Contains(curr, ";") {
-					processCollectedSql(path, &singleLine, &singleString, objType, &sqlInfoArr, &reportNextSql)
-					//reset for parsing other sqls
-					dollarQuoteFlag = 0
-					outerDollarQuotedStr = ""
-				}
+		} else if dollarQuoteFlag == 1 {
+			if strings.Contains(curr, outerDollarQuotedStr) {
+				dollarQuoteFlag = 2 //denotes end of code/body part
```

Comment:
We can get rid of the `dollarQuoteFlag == 2` case by moving block from line 593-599 to line 589.
That will leave only two valid values, 0 and 1, for the `dollarQuoteFlag`. In that case, we can change its type to `bool` and perhaps name it as `insideCodeBlock`.
Then you can simplify the if..else block as:
```
if !insideCodeBlock {

} else {
    // insideCodeBlock
}
```

Reply:
No we can't do that, i have thought of this. But sometimes there are some clause like `LANGUAGE PL/PGSQL` after the ending of Code Block and before the end of functions stmt. Basically multiple lines are possible after Code block but before ending of function 

So that is covered in current way of implemented but not this.

Reply:
Got it! 👍